packages/manager/modules/exchange/src/account/add/account-add.controller.js

```diff
@@ -214,21 +248,46 @@ export default class ExchangeAccountAddController {
         formattedAccount,
       )
       .then((data) => {
-        this.messaging.writeSuccess(
-          this.$translate.instant('exchange_account_add_submit_success', {
-            t0: `${formattedAccount.login}@${formattedAccount.domain}`,
-          }),
-          data,
+        this.goToAccounts().then(() =>
+          this.messaging.writeSuccess(
+            this.$translate.instant('exchange_account_add_submit_success', {
+              t0: `${formattedAccount.login}@${formattedAccount.domain}`,
+            }),
+            data,
+          ),
         );
       })
       .catch((error) => {
         this.messaging.writeError(
-          this.$translate.instant('exchange_ACTION_add_account_error_message'),
+          this.$translate.instant('exchange_ACTION_add_account_error_message', {
+            message: error.message || error.data?.message,
+          }),
           error,
         );
       })
       .finally(() => {
-        this.hide();
+        this.isSendingNewAccount = false;
+
+        this.$location.hash(EXCHANGE_CONTAINER_MESSAGING);
+        this.$anchorScroll();
```

Comment:
Hmm shouldn't this be only done for `catch` case? As you are making state transition when promise succeeds, the anchor scroll won't be effective

Reply:
It's applied